core/test_network/src/lib.rs

```diff
@@ -81,71 +75,44 @@ pub trait TestGenesis: Sized {
 impl TestGenesis for GenesisNetwork {
     fn test(submit_genesis: bool) -> Option<Self> {
         let cfg = Configuration::test();
-        let mut genesis = RawGenesisBlock::new(
-            "alice".parse().expect("Valid"),
-            "wonderland".parse().expect("Valid"),
-            get_key_pair().public_key().clone(),
-        );
+
+        // TODO: Fix this somehow. Probably we need to make `kagami` a library.
+        let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let content = std::fs::read_to_string(manifest_dir.join("../../configs/peer/genesis.json"))
+            .expect("Failed to read data from configs/peer/genesis.json");
+        let mut genesis: RawGenesisBlock =
+            json5::from_str(&content).expect("Failed to deserialize genesis block from config");
+
         let rose_definition_id = <AssetDefinition as Identifiable>::Id::from_str("rose#wonderland")
             .expect("valid names");
         let alice_id =
             <Account as Identifiable>::Id::from_str("alice@wonderland").expect("valid names");
-        let mint_rose_permission: PermissionToken =
-            CanMintUserAssetDefinitions::new(rose_definition_id.clone()).into();
-        let burn_rose_permission: PermissionToken =
-            CanBurnAssetWithDefinition::new(rose_definition_id.clone()).into();
 
-        genesis.transactions[0].isi.extend(
-            public_blockchain::default_permission_token_definitions()
-                .into_iter()
-                .chain(private_blockchain::default_permission_token_definitions().into_iter())
-                .map(|token_definition| RegisterBox::new(token_definition.clone()).into()),
-        );
-        genesis.transactions[0].isi.push(
-            RegisterBox::new(AssetDefinition::quantity(
-                AssetDefinitionId::from_str("rose#wonderland").expect("valid names"),
-            ))
-            .into(),
-        );
+        // TODO: Probably use strongly-typed permission from validators crates
```

Comment:
is it possible to use strongly typed permissions?

Reply:
Unfortunately, no. See [this](https://github.com/hyperledger/iroha/pull/3106#discussion_r1114217036) comment.

I don't think it's impossible. But I don't think we need to work on this non-trivial thing for deprecated tests. Better to do stuff for future Python tests

Reply:
so I thought, then remove the TODO